src/viur/core/bones/phone.py

```diff
@@ -0,0 +1,129 @@
+import re
+import typing as t
+import json
+
+from viur.core.bones.string import StringBone
+from viur.core.bones.base import ReadFromClientError, ReadFromClientErrorSeverity
+from viur.core import conf
+
+from pathlib import Path
+
+DEFAULT_REGEX = r"^\+?(\d{1,3})[-\s]?(\d{1,4})[-\s]?(\d{1,4})[-\s]?(\d{1,9})$"
+
+COUNTRYFILE = Path(__file__).parent / "../languages/country_information.json"
+
+with open(COUNTRYFILE, "r") as fin:
+    COUNTRYDATA = json.load(fin)
+
+
+class PhoneBone(StringBone):
+    """
+    The PhoneBone class is designed to store validated phone/fax numbers in configurable formats.
+    This class provides a number validation method, ensuring that the given phone/fax number conforms to the
+    required/configured format and structure.
+    """
+
+    type: str = "str.phone"
+    """
+    A string representing the type of the bone, in this case "str.phone".
+    """
+
+    def __init__(
+        self,
+        *,
+        test: t.Optional[t.Pattern[str]] = DEFAULT_REGEX,
+        max_length: int = 15,  # maximum allowed numbers according to ITU-T E.164
+        default_country_code: t.Optional[str] = None,
+        **kwargs: t.Any,
+    ) -> None:
+        """
+        Initializes the PhoneBone with an optional custom regex for phone number validation, a default country code,
+        and a flag to apply the default country code if none is provided.
+
+        :param test: An optional custom regex pattern for phone number validation.
+        :param max_length: The maximum length of the phone number. Passed to 'StringBone'.
+        :param default_country_code: The default country code to apply if none is provided.
+        :param kwargs: Additional keyword arguments. Passed to 'StringBone'.
+        """
+        self.test: t.Pattern[str] = re.compile(test) if isinstance(test, str) else test
+        self.default_country_code: str = default_country_code
```

Comment:
The parameter can be `None` or a `str`, but the instance variable is only `str`. How?